src/include/miopen/convolution.hpp

```diff
@@ -437,6 +456,23 @@ struct ConvolutionDescriptor : miopenConvolutionDescriptor
                               miopenConvSolution_t* solutions) const;
 
     std::size_t GetSolutionCountFallback(Handle& handle, const ProblemDescription& problem) const;
+
+    template <class Stream, std::enable_if_t<IsBinarySerializationRelated<Stream>{}, bool> = true>
+    friend Stream& operator<<(Stream& stream, ConvolutionDescriptor& conv)
```

Comment:
If this is serializing then `ConvolutionDescriptor` can be `const`. That is why the `Visit` takes templated `Self` parameter so it can be used for both.